packages/rpc-transport/src/apis/api-types.ts

```diff
@@ -0,0 +1,11 @@
+export type RpcApiConfig = Readonly<{
+    parametersTransformer?: <T>(params: T, methodName: string) => unknown[];
+    responseTransformer?: <T>(response: unknown, methodName: string) => T;
+}>;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type RpcMethod = (...args: any) => any;
+
+export interface IRpcApiMethods {
```

Comment:
Singular? `IRpcApiMethod`?

Reply:
The interface itself can have a bunch of overloads. But if we consider those to be all one method (they are), then yeah I guess we can do singular.

Reply:
Actually it works if you just roll the whole type spec into one interface, too. So maybe plural is OK?

<img width="758" alt="Screenshot 2023-11-15 at 1 35 27 PM" src="https://github.com/solana-labs/solana-web3.js/assets/67205063/f061b660-dc05-4bd4-b9ec-2b76c8c19012">

Reply:
It's a tricky one because `IRpcApiMethods` best describe the type (as you can bundle multiple methods in one API type) but we do want to encourage people to provide one type per method so users can cherry-pick the methods they require. I'm fine with either but I have a slight preference towards to the plural version.